src/main/java/frc/robot/subsystems/ColorSensor.java

```diff
@@ -0,0 +1,58 @@
+package frc.robot.subsystems;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.GameConstants;
+import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
+
+public class ColorSensor extends SubsystemBase{
+    private final I2C.Port i2cPort = I2C.Port.kOnboard;
+    private final ColorSensorV3 m_colorSensor = new ColorSensorV3(i2cPort);
+    private final ColorMatch m_colorMatcher = new ColorMatch();
+    String colorObject;
+    public ColorSensor() {
+        m_colorMatcher.addColorMatch(GameConstants.C_PIECE_TARGET);
+        m_colorMatcher.addColorMatch(GameConstants.C_PLASTIC_TARGET);
+    }
+    @Override
+    public void periodic() {
+        Color detectedColor = m_colorSensor.getColor();
+        double IR = m_colorSensor.getIR();
+        
+        double proximity = m_colorSensor.getProximity();
+        
+        ColorMatchResult matchedColor = m_colorMatcher.matchClosestColor(detectedColor);
+        if (matchedColor.color == GameConstants.C_PIECE_TARGET) {
+            colorObject="Piece";
+        } else if (matchedColor.color == GameConstants.C_PLASTIC_TARGET) {
+            colorObject="Plastic";
+        } else {
+            colorObject="None";
```

Comment:
These color object strings should be made an enum instead

Reply:
MAde into an enum and created the following switch statement needed

Reply:
see my new comment about the enums